extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts

```diff
@@ -0,0 +1,166 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
```

Comment:
use `import type * as azdataType from 'azdata'`, like in publishDatabaseDialog.ts. This is needed so there aren't any errors in the sql projects vscode extension. Even thought this dialog isn't used in vscode, there would still be an error that azdata can't be found

Reply:
When I use type, It is suggesting to remove the type as "'azdata' cannot be used as a value because it was imported using 'import type'." for azdata.window.createTab().

Reply:
You should never use `import * as azdata` in sql DB proj extension since as @kisantia mentioned above this will break the VS code version of the extension (even if this code is never called directly in that scenario, the imports alone will break it at runtime)

You need to use either : 

*  `import type * as azdata` if you are ONLY using the typings (no objects, including functions. So in your example you can't use this one)
* Use [getAzdataApi()](https://github.com/microsoft/azuredatastudio/blob/d95aff1d3c89e867ad925b357b3598bb156f62c8/extensions/sql-database-projects/src/common/utils.ts#L418) if you need to use actual functionality from the extension

Reply:
clear!